src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -2890,6 +2890,17 @@ private Socket getConnectedSocket(InetSocketAddress addr, int timeoutInMilliSeco
         if (addr.isUnresolved())
             throw new java.net.UnknownHostException();
         selectedSocket = getSocketFactory().createSocket();
+
+        String localSocketAddress = this.conn.activeConnectionProperties.getProperty(SQLServerDriverStringProperty.LOCAL_SOCKET_ADDRESS.toString());
+
+        if (null != localSocketAddress && !localSocketAddress.isEmpty()) {
+            selectedSocket.bind(new InetSocketAddress(InetAddress.getByName(localSocketAddress), 0));
```

Comment:
Do we validate this user specified loicalSocketAddress as a valid IP somewhere before coming here ? If not would it be good to handle an exception thrown by InetSocketAddress/bind call and raise an error in that case ?